### PR TITLE
rustup: new, 1.27.1

### LINKED ISSUE #16145: Add rustup
### Package name

rustup

### Brief description of package

rustup is an installer for
 the systems programming language Rust

### URL for package's homepage

https://rustup.rs/

### Provide a basic test case to validate the package's functionality.

_No response_

### MINGW environments where you need the package

- [X] MINGW64
- [ ] MINGW32
- [X] UCRT64
- [X] CLANG64
- [ ] CLANG32
- [X] CLANGARM64

### Are you willing to submit a PR?

_No response_

### PR DESCRIPTION
**Pull request by 500-internal-server-error**
Fixes #16145

### CODE REVIEW

**Comment by ognevny** [comment]
why ucrt64 only?

**Comment by ognevny** [comment]
fix rustup-profile.sh to use ${MINGW_PREFIX}

File: mingw-w64-rustup/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,66 @@
+_realname=rustup
+pkgbase=mingw-w64-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+pkgver=1.27.1
+pkgrel=1
+pkgdesc="The Rust toolchain installer"
```
Comment (filnet) [inline mingw-w64-rustup/PKGBUILD:6]: Description should have a trailing `  (mingw-w64)`.

File: mingw-w64-rustup/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	"${MINGW_PACKAGE_PREFIX}-gdb: rust-gdb script"
+	"${MINGW_PACKAGE_PREFIX}-lldb: rust-lldb script"
+)
+provides=("${MINGW_PACKAGE_PREFIX}-rust")
+conflicts=("${MINGW_PACKAGE_PREFIX}-rust")
+install='post.install'
+options=("!lto")
+source=(
+	"rustup-${pkgver}.tar.gz::${url}/archive/${pkgver}.tar.gz"
+	"rustup-profile.sh"
```
Comment (filnet) [inline mingw-w64-rustup/PKGBUILD:30]: why is `post.install` not validated like `rustup-profile.s` is ?

File: mingw-w64-rustup/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+license=('spdx:Apache-2.0 AND MIT')
+msys2_references=(
+	'archlinux: rustup'
+)
+depends=(
+	"${MINGW_PACKAGE_PREFIX}-curl"
+	"${MINGW_PACKAGE_PREFIX}-xz"
+	"${MINGW_PACKAGE_PREFIX}-zstd"
+)
+makedepends=("${MINGW_PACKAGE_PREFIX}-rust")
```
Comment (filnet) [inline mingw-w64-rustup/PKGBUILD:19]: So the MSYS2 rust is needed to build rustup.
And the 1st thing rustup does it kick out MSYS2 rust.
Not very polite :)

File: mingw-w64-rustup/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,66 @@
+_realname=rustup
+pkgbase=mingw-w64-${_realname}
+pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
+pkgver=1.27.1
+pkgrel=1
+pkgdesc="The Rust toolchain installer"
+arch=('x86_64')
+mingw_arch=('mingw64' 'ucrt64' 'clang64' 'clangarm64')
+url="https://github.com/rust-lang/rustup.rs"
```
Comment (filnet) [inline mingw-w64-rustup/PKGBUILD:9]: The `.rs` extension seems unnecessary.

File: mingw-w64-rustup/PKGBUILD (hunk 1 above)
Comment (500-internal-server-error) [reply mingw-w64-rustup/PKGBUILD]: Added

File: mingw-w64-rustup/PKGBUILD (hunk 2 above)
Comment (500-internal-server-error) [reply mingw-w64-rustup/PKGBUILD:30]: I mostly followed Arch's `PKGBUILD`, they don't validate it either.

File: mingw-w64-rustup/PKGBUILD (hunk 4 above)
Comment (500-internal-server-error) [reply mingw-w64-rustup/PKGBUILD]: I mostly followed Arch's `PKGBUILD`, I would guess it's just the old link that never got updated. I updated it for this PR.

File: mingw-w64-rustup/PKGBUILD (hunk 3 above)
Comment (500-internal-server-error) [reply mingw-w64-rustup/PKGBUILD:19]: :) It builds too if an existing rust install can be found elsewhere (e.g., manual download of rustup-init), with `--nodeps`.

**Comment by 500-internal-server-error** [comment]
Bump :) Is there anything else I need to do?

**Comment by ognevny** [comment]
why did you reset your commit?

**Comment by 500-internal-server-error** [comment]
Superseded by #20989

Forgot to do the work in a separate branch, updating master on my fork caused merge conflicts :)

**Comment by 500-internal-server-error** [comment]
> why did you reset your commit?

I tried to sync my fork with this repo, which failed with a merge conflict. Because I forgot to add my commits to a separate branch, I cannot my fork's master with this repo's master without rewriting history.

**Pull request closed without merging**